pkg/operator/operator_config.go

```diff
@@ -485,10 +484,10 @@ func (r *operatorConfigReconciler) ensureAlertmanagerConfigSecret(ctx context.Co
 }
 
 type alertmanagerConfig struct {
```

Comment:
Could we take advantage of [this](https://github.com/prometheus/alertmanager/blob/9b35e7696449df7e99c97a6721daeb3f500f8242/config/config.go#L42-L45) to allow umarshalling of secrets when required instead of cloning the upstream config? 

```
// MarshalSecretValue if set to true will expose Secret type
// through the marshal interfaces. Useful for outside projects
// that load and marshal the Alertmanager config.
var MarshalSecretValue bool = commoncfg.MarshalSecretValue
```
IE.  we can set `promcommonconfig.MarshalSecretValue = true` when needed